mongoclient/db.go

```diff
@@ -0,0 +1,106 @@
+package mongoclient
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"go.mongodb.org/mongo-driver/mongo/readpref"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/netlify/netlify-commons/nconf"
+)
+
+const (
+	CollectionBlobs         = "blobs"
+	CollectionResellers     = "resellers"
+	CollectionUsers         = "users"
+	CollectionSubscriptions = "bb_subscriptions"
+	CollectionSites         = "projects"
+)
+
+type Auth struct {
+	Username string
+	Password string
+	Source   string
+}
+
+type Config struct {
+	AppName            string
+	TLS                *nconf.TLSConfig
+	Servers            []string
+	ReplSetName        string
+	ConnTimeout        time.Duration
+	Auth               *Auth
+	SecondaryPreferred bool
+}
+
+// Connect connects to MongoDB
+func Connect(config *Config, log *logrus.Entry) (*mongo.Client, error) {
```

Comment:
What do you think about using a `Option` driven API (like the `router` package).

Something like 
```
type Option func(opt *options.ClientOptions) error

func Connect(log logrus.FieldLogger, replSet string, servers []string, opts ...Option) (*mongo.Client, error) {
	finalOpts := options.Client().
		SetConnectTimeout(time.Minute). // use a default
		SetReplicaSet(replSet).
		SetHosts(servers)
	for _, opt := range opts {
		if err := opt(finalOpts); err != nil {
			return nil, err
		}
	}
	log.WithFields(logrus.Fields{
		"servers":     strings.Join(finalOpts.Hosts, ","),
		"replica_set": finalOpts.ReplicaSet,
	}).Debug("Dialing database")

	client, err := mongo.Connect(context.Background(), finalOpts)
	if err != nil {
		return nil, err
	}

	// Connect does not block for server discovery, so we should ping to ensure
	// we are actually connected
	ctx, cancel := context.WithTimeout(context.Background(), *finalOpts.ConnectTimeout+time.Second)
	defer cancel()

	if err := client.Ping(ctx, readpref.Primary()); err != nil {
		log.WithError(err).Error("Failed to ping primary Mongo instance")
		return nil, err
	}

	log.Debug("Connected to MongoDB")
	return client, nil
}

func TLSOption(config *nconf.TLSConfig) Option {
	return func(opts *options.ClientOptions) error {
		tlsConfig, err := config.TLSConfig()
		if err != nil {
			return err
		}
		opts.SetTLSConfig(tlsConfig)
		return nil
	}
}

func AuthOption(config Auth) Option {
	return func(opts *options.ClientOptions) error {
		creds := options.Credential{
			Username:   config.Username,
			Password:   config.Password,
			AuthSource: config.Source,
		}

		opts.SetAuth(creds)
		return nil
	}
}
```

It would expose the future direct mongo options too so that we don't have to tweak the API a lot. 

One drawback is that you can't directly import `Config` struct and just start using it. (though we could have 2 methods `ConfigWithOptions` and `Config`)

Reply:
I like that idea. Let me open an issue for that as that's somewhat a departure from the previous API, but I think a good direction to take.